### PR TITLE
Fix for correctly interpreting the whitelist under Windows

### PR DESCRIPTION
**Pull request by jeremypoulter**
os.path.normcase also converts / to \ on Windows (using the native Windows Python). This caused the paths in the white list (that have been run through normcase) to not match the file paths (that have not been run through normpath). This fix just uses the normpath version of the filepath for matching rather than the original path.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/wpt-tools/142)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Did this cause test failures on Windows?

**Comment by gsnedders** [comment]
@Ms2ger No, because otherwise master would be failing.

@jeremypoulter Can you please write a test (in `lint/tests/test_lint.py` that fails on master and passes on this branch?

**Comment by gsnedders** [comment]
Relatedly, why on earth has AppVeyor not run this?

**Comment by jeremypoulter** [comment]
As far as I could see AppVeyor runs under MSys so is probably using a
different port of Python.

I will try and create a test for this.

Jeremy

On 6 Dec 2016 10:36 p.m., "Geoffrey Sneddon" <notifications@github.com>
wrote:

Relatedly, why on earth has AppVeyor not run this?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/w3c/wpt-tools/pull/142#issuecomment-265295422>, or mute
the thread

**Comment by jeremypoulter** [comment]
Not sure where I got the MSys from, it was a few weeks ago when I fixed this. However the issue is not showing on AppVeyor is I do not think that test_filter_whitelist_errors does not really test filter_whitelist_errors where the error is. I am currently writing some more extensive testing.

**Comment by jeremypoulter** [comment]
I have added tests for filter_whitelist_errors that shows errors on the master branch that are fixed with the PR

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+*	-text
```
Comment (gsnedders) [inline .gitattributes]: Can we just set this on `lint/tests/dummy/**`?

Comment (jeremypoulter) [inline .gitattributes]: I guess for this repository this is ok, however this change as is replicates what AppVeyor is doing; http://help.appveyor.com/discussions/problems/671-coreautocrlf.

Comment (gsnedders) [inline .gitattributes]: We're only broken with `core.autocrlf=true`, right? IMO we should respect people's git configurations and only deviate when we actually have to (which is the test files only).

Probably worth setting `core.autocrlf` to true on AppVeyor to make sure we don't break under such a configuration.

Comment (jeremypoulter) [inline .gitattributes]: Is it expected that lint should be run on the files in this repository or just web-platform-test? if the former then lint enforces LF so I think what I have is correct if the latter then I think it is ok to limit the change to lint/tests/dummy/*

Comment (jeremypoulter) [inline .gitattributes]: I will also look at what is needed to set AppVeyor to use core.autocrlf=true

Comment (gsnedders) [inline .gitattributes]: The latter. The only time lint is run on this repo is as part of the tests.

Pretty sure all you need for AppVeyor is `init: git config --global core.autocrlf true` though I could be wrong.

Comment (gsnedders) [inline .gitattributes]: https://github.com/jeremypoulter/wpt-tools/pull/1 fixes this.

**Pull request merged**